app/validators/questionnaire_schema.py

```diff
@@ -412,9 +412,18 @@ def _get_numeric_value_for_value_source(
             referred_answer = answer_ranges.get(defined_value["identifier"])
         elif value_source == "calculated_summary":
             calculated_summary_block = self.get_block(defined_value["identifier"])
-            answers_to_calculate = calculated_summary_block["calculation"][
-                "answers_to_calculate"
-            ]
+            if calculated_summary_block["calculation"].get("answers_to_calculate"):
+                answers_to_calculate: list = calculated_summary_block["calculation"][
+                    "answers_to_calculate"
+                ]
+            else:
+                answers_to_calculate = [
+                    value["identifier"]
+                    for value in calculated_summary_block["calculation"]["operation"][
+                        "+"
+                    ]
+                    if value["source"] == "answers"
+                ]
```

Comment:
This is essentially a duplicate of `get_calculated_answer_ids`. I suggest rewriting that to work with a given schema. Then you can reuse here.